### DISCUSSION TITLE
hide and show texture

### DISCUSSION THREAD

**Discussion by HasmikOsensus** [google/model-viewer#3976] [General]
how to show and hide the model's texture by react js?
hiding the texture works correctly. Cant handle returning the texture back part

**Comment by HasmikOsensus** [comment]
this is the code 
const handleTextureChange =  (e) => {
    setIsTexturedChecked((prevState) => !prevState);
    const modelViewer = document.getElementById("3d-model-viewer");
    modelViewer.addEventListener("load", () => {
      const material = modelViewer.model.materials[0];
      const texture = material.pbrMetallicRoughness.baseColorTexture.texture.source;
     if (e) {
         material.pbrMetallicRoughness.baseColorTexture.setTexture(null);
        
      } else {
        material.baseColorTexture.setTexture(texture);
      }
    });
  };

**Reply by elalish** [reply]
Try removing `.source`: the texture is just `baseColorTexture.texture`. Also, it's a bit hard to read your code, but it looks like your functions are mixed up with each other.
